runtime/runtime/src/lib.rs

```diff
@@ -1499,10 +1500,70 @@ impl Runtime {
             prefetcher.clear();
         }
 
+        // Resolve timed-out yielded promises
```

Comment:
style: should perhaps consider splitting this out into a function. The `apply` is quite a bit too large as is.

Reply:
Makes sense, I will look into sending a follow-up PR refactoring `apply`.